hedera-node/hedera-mono-service/src/main/java/com/hedera/services/context/MutableStateChildren.java

```diff
@@ -231,7 +242,7 @@ public void updateFrom(final ServicesState state) {
     }
 
     public void updatePrimitiveChildrenFrom(final ServicesState state) {
-        accounts = new WeakReference<>(state.accounts());
+        accounts = new NonAtomicReference<>(state.accounts());
```

Comment:
Before, the `MerkleMap` target of this reference was _also_ a child of a `ServicesState` copy. As long as that copy was not released, the `ServicesState` reference would prevent the `MerkleMap` from being garbage collected.

But now the `AccountsStorageAdapter` returned [from `ServicesState.accounts()`](https://github.com/hashgraph/hedera-services/blob/virtualize-more-maps/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ServicesState.java#L447) may have no other references other than the one held by this `MutableStateChildren`.

So we need a strong reference to this `AccountStorageAdapter` to prevent it from being garbage collected before we are done with it.